xdsl/context.py

```diff
@@ -108,11 +119,20 @@ def load_op(self, op: "type[Operation]") -> None:
             raise Exception(f"Operation {op.name} has already been loaded")
         self._loaded_ops[op.name] = op
 
-    def load_attr(self, attr: "type[Attribute]") -> None:
-        """Load an attribute definition. Attribute names should be unique."""
-        if attr.name in self._loaded_attrs:
-            raise Exception(f"Attribute {attr.name} has already been loaded")
-        self._loaded_attrs[attr.name] = attr
+    def load_attr_or_type(self, attr: type[Attribute]) -> None:
+        """
+        Load an attribute or type definition.
+        Attribute or type names should be unique, but an attribute may have the same
+        name as a type.
+        """
+        if issubclass(attr, TypeAttribute):
+            if attr.name in self._loaded_types:
+                raise Exception(f"Type {attr.name} has already been loaded")
```

Comment:
is there a better error here? At least a ValueError IMO, but maybe a custom subclass?

Reply:
I added a new one